Neos.Flow/Classes/Http/Middleware/SecurityEntryPointMiddleware.php

```diff
@@ -0,0 +1,100 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\Flow\Http\Middleware;
+
+use GuzzleHttp\Psr7\Response;
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Http\ServerRequestAttributes;
+use Neos\Flow\Log\Utility\LogEnvironment;
+use Neos\Flow\Mvc\ActionRequestFactory;
+use Neos\Flow\Security\Authentication\Token\SessionlessTokenInterface;
+use Neos\Flow\Security\Authentication\TokenInterface;
+use Neos\Flow\Security\Context;
+use Neos\Flow\Security\Exception\AuthenticationRequiredException;
+use Psr\Log\LoggerInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+/**
+ * A HTTP middleware that handles authentication exceptions that were thrown by the dispatcher (@see \Neos\Flow\Mvc\Dispatcher::dispatch()) and
+ * * rethrows the exception if no token with Entry Point is authenticated
+ * * or otherwise invokes the Entry Point of all authenticated tokens
+ */
+class SecurityEntryPointMiddleware implements MiddlewareInterface
+{
+    /**
+     * @Flow\Inject(lazy=false)
+     * @var Context
+     */
+    protected $securityContext;
+
+    /**
+     * @Flow\Inject(name="Neos.Flow:SecurityLogger")
+     * @var LoggerInterface
+     */
+    protected $securityLogger;
+
+    /**
+     * @Flow\Inject(lazy=false)
+     * @var ActionRequestFactory
+     */
+    protected $actionRequestFactory;
+
+    /**
+     * @inheritDoc
+     */
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $next): ResponseInterface
+    {
+        try {
+            // FIXME: Currently the security context needs an ActionRequest, therefore we need to build it here
+            $routingMatchResults = $request->getAttribute(ServerRequestAttributes::ROUTING_RESULTS) ?? [];
```

Comment:
As discussed with @bwaidelich - the SecurityContext right now has an unfortunate dependency on the `ActionRequest`. Since this middleware wraps the security concept of the framework, and the `ActionRequest` is not really supposed to exist within the PSR middleware, this is the place to create it. That way, we do not reintroduce a middleware that creates an ActionRequest that 3rd party could depend on. The `ServerRequestAttributes::ACTION_REQUEST` is marked internal for this matter.

That's also why the `PrepareMvcRequestComponent` is removed with this now.